### ISSUE TITLE
macOS client requires Firefox as default browser

### ISSUE THREAD

**Issue by pronouncedJerry** [mozilla-mobile/mozilla-vpn-client#2846]
For the past week my VPN client on macOS 12.3 has not been connecting. I reached out to Support for help and I am a bit baffled by the response to my issue so I would like to confirm with the development team that this is an official stance by Mozilla:

> Could you please make sure that you login via the Firefox browser in order to set up the VPN successfully? It seems like other users encountered the same problem on MacOS and it worked after setting Firefox as their default browser when logging in.

Firefox is now required to be the default browser on macOS in order for the VPN client to work?

Since I've been unable to use this service I pay for and kindly request that this new requirement be made clear to new customers so they don't fall into the same trap I did. Otherwise, it seems like a prevalent problem (and top priority) that should be fixed, no?

**Comment by oskirby** [comment]
I agree that this should be fixed. Our support team will often request trying other browsers as a troubleshooting step, but we should expect login and authentication be supported on all the major browsers. Would you be able to share the browser, and its version, that you were trying to use? If you're willing to provide logs from the VPN client after your login attempt, that would be appreciated too.

**Comment by pronouncedJerry** [comment]
Thank you for the quick response oskirby.

Safari version 15.4 (build 17613.1.14.1.7)

[mozillavpn-2022-2-23_2.txt](https://github.com/mozilla-mobile/mozilla-vpn-client/files/8128651/mozillavpn-2022-2-23_2.txt) hopefully I removed all of my PII

**Comment by oskirby** [comment]
Alright, I have been able to reproduce the issue on Safari 15.2 (17612.3.6.1.6). It seems the issue here is that Safari insists on forcing the use of HTTPS when performing the OAuth redirect to localhost. For some technical info as to why, and some really gross workarounds, see: [here](https://stackoverflow.com/questions/46394682/safari-keeps-forcing-https-on-localhost).

**Comment by pronouncedJerry** [comment]
The issue referenced on SO originated 4 years ago. The issue with the vpn client start a week ago, though.

**Comment by birdsarah** [comment]
@oskirby as mentioned earlier, we use [deeplinks for some mobile cases on guardian](https://github.com/mozilla-services/guardian-website/blob/releases/1.21/server/lib/auth.ts#L60) (private repo link). This documentation from apple is making me hopeful we could implement for mac: https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app but i'm not sure if it implies delivery through the mac store.

**Comment by birdsarah** [comment]
> The issue referenced on SO originated 4 years ago. The issue with the vpn client start a week ago, though.

@pronouncedJerry I don't have an immediate answer for this. Authentication code didn't change in the latest VPN client release to my knowledge. We are certainly looking into this though.

**Comment by pronouncedJerry** [comment]
hello. is there an update available on this issue? where on the priority list for bugs does this sit? I have not been able to use the macOS client for over a month now.

**Comment by oskirby** [comment]
@pronouncedJerry We are planning to address this in the next release of the VPN client (version 2.8) that should resolve this problem, and allow users to login using Safari as their default browser. We hope to have this update available in early April as long as things go smoothly.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Guardian 1.24 is released in stage.

We verified this as fixed while using Mozilla VPN 2.8.3 (2.202205040930) on Mac OS 10.15 while having the default browser set to Safari.

The authentication from VPN works as expected.

There is a popup from Safari that is triggered after log in as you can see below. If the user chooses “Cancel“ the webpage says “Success“ and the client remains in loading state. Is this client correct behavior? Andrea Marchesini 

!Screenshot 2022-05-05 at 12.21.31.png|width=818,height=816!

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Baku confirmed this is the intended behavior.

**Comment by pronouncedJerry** [comment]
Huh @ValentinaPC why would the intended behavior be to stay in the loading state? that's been the whole issue since 3 months ago. when is this going to get resolved? I haven't been able to use the client on my laptop.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Because that popup is from Safari (security setting) and we do not have control of it. If you choose “Allow“ you will be logged into Mozilla VPN client. If you chose “Cancel“ that means that you block this.
